### ISSUE TITLE
Multiple tracks - next track not starting on mobile

### ISSUE THREAD

**Issue by Gigamick** [katspaugh/wavesurfer.js#2656]
I have a site that loads a dynamic number of tracks onto the page.

I have code that plays the next track when the current track ends:

```
   wavesurfer{{$p}}.on('finish', function () {

        if (typeof wavesurfer{{$p+1}}) 
            document.getElementById("sb_playbutton{{$p+1}}").click();
    })
```

It clicks the play button of the NEXT wavesurfer. This works fine on desktop but on mobile it doesn't. I'd love some steer here.

Using version 6.2.0

**Comment by prakharpbuf** [comment]
When do you call the wavesurfer.load(<AUDIO>) on these wavesurfer instances?

**Comment by Gigamick** [comment]
> When do you call the wavesurfer.load(<AUDIO>) on these wavesurfer instances?
> 
> Mobile phones generally don't allow you to automatically play (actually load) audio without any user interaction first.

Yeah this is my fear. So how do you go around making several tracks okay back to back? This seems like a very common use case - press play on track one and listen to tracks 2, 3, 4 etc without needing to press play on every one.

**Comment by prakharpbuf** [comment]
With the limited information I have about your code and use case, I assume you are calling the wavesurfer.load() somewhere in the onclick handler of your play button.

Try:

```
 wavesurfer{{$p}}.on('finish', function () {
      if (typeof wavesurfer{{$p+1}}) {
            wavesurfer{{$p+1}}.load(<AUDIO>)
            wavesurfer{{$p+1}}.on('ready', () => wavesurfer{{$p+1}}.play());
      }
  })
```

This _might_ work.
Please try it out and lemme know if it worked or not

**Comment by prakharpbuf** [comment]
Aside: I was also using multiple wavesurfer instances in one of my pages for testing. I found that storing these wavesurfer instances in an array, or even better, a json object with meaningful key names made it much easier to work with.

**Comment by Gigamick** [comment]
So the code snippet you gave me didn't work, but it did lead me to this which works great!

```
  if (typeof wavesurfer{{$p+1}}) {
                wavesurfer{{$p+1}}.play();
            }
```

Seems like calling the play function is fine but trying to simulate a click (which is what my original code was doing) was a red flag.

Thank you 🙏


> With the limited information I have about your code and use case, I assume you are calling the wavesurfer.load() somewhere in the onclick handler of your play button.
> 
> Try:
> 
> ```
>  wavesurfer{{$p}}.on('finish', function () {
>       if (typeof wavesurfer{{$p+1}}) {
>             wavesurfer{{$p+1}}.load(<AUDIO>)
>             wavesurfer{{$p+1}}.on('ready', () => wavesurfer{{$p+1}}.play());
>       }
>   })
> ```
> 
> This _might_ work. Please try it out and lemme know if it worked or not
